pkg/policy/validate.go

```diff
@@ -425,6 +422,39 @@ func Validate(policy kyvernov1.PolicyInterface, client dclient.Interface, mock b
 	return warnings, nil
 }
 
+func UpdateSourceResource(client dclient.Interface, kind, namespace string, policy kyvernov1.PolicyInterface, obj *unstructured.Unstructured) error {
+	updateSource := true
+	label := obj.GetLabels()
+
+	if len(label) == 0 {
+		label = make(map[string]string)
+		label["generate.kyverno.io/clone-policy-name"] = policy.GetName()
+	} else {
+		if label["generate.kyverno.io/clone-policy-name"] != "" {
+			policyNames := label["generate.kyverno.io/clone-policy-name"]
+			if !strings.Contains(policyNames, policy.GetName()) {
+				policyNames = policyNames + "," + policy.GetName()
+				label["generate.kyverno.io/clone-policy-name"] = policyNames
+			} else {
+				updateSource = false
+			}
+		} else {
+			label["generate.kyverno.io/clone-policy-name"] = policy.GetName()
+		}
+	}
+
+	if updateSource {
+		logging.V(4).Info("updating existing clone source labels")
+		obj.SetLabels(label)
```

Comment:
You may need to reset the resourceVersion to empty to avoid the "object has been modified" error.
```
obj.SetResourceVersion("")
```

Reply:
good idea !